core/src/main/java/tech/ydb/core/impl/operation/OperationManager.java

```diff
@@ -0,0 +1,194 @@
+
+    private final GrpcTransport transport;
+    private final ScheduledExecutorService scheduledExecutorService;
+    private final GrpcRequestSettings requestSettings = GrpcRequestSettings.newBuilder().build();
+
+    public OperationManager(GrpcTransport transport) {
+        this.transport = transport;
+        this.scheduledExecutorService = transport.getScheduler();
+    }
+
+    public <Value extends Message> Operation<Value> createOperation(
+            final OperationProtos.Operation operationProto,
+            final Class<Value> resultClass
+    ) {
+        Operation<Value> operation = new Operation<>(
+                operationProto.getId(),
+                resultClass,
+                this
+        );
+
+        completeOperation(operationProto, operation);
+
+        return operation;
+    }
+
+
+    private <Value extends Message> void completeOperation(
+            final OperationProtos.Operation operationProto,
+            final Operation<Value> operation
+    ) {
+        if (operation.resultCompletableFuture.isCancelled()) {
+            return;
+        }
+
+        final Status status = status(operationProto);
+
+        if (operationProto.getReady()) {
+            if (status.isSuccess()) {
+                try {
+                    Value resultMessage = operationProto
+                            .getResult()
+                            .unpack(operation.resultClass);
+                    operation.resultCompletableFuture
+                            .complete(Result.success(resultMessage, status));
+                } catch (InvalidProtocolBufferException ex) {
+                    operation.resultCompletableFuture.complete(
+                            Result.error(
+                                    "Can't unpack message " + operation.resultClass.getName(),
+                                    ex
+                            )
+                    );
+                }
+            } else {
+                operation.resultCompletableFuture.complete(Result.fail(status));
+            }
+
+            return;
+        }
+
+        scheduledExecutorService.schedule(
+                () -> {
+                    OperationProtos.GetOperationRequest request = OperationProtos.GetOperationRequest
+                            .newBuilder()
+                            .setId(operation.operationId)
+                            .build();
+
+                    transport.unaryCall(
+                            OperationServiceGrpc.getGetOperationMethod(),
+                            requestSettings,
+                            request
+                    ).whenComplete(
+                            (getOperationResponseResult, throwable) -> {
+                                if (throwable != null) {
+                                    operation.resultCompletableFuture.completeExceptionally(throwable);
+                                } else if (getOperationResponseResult != null) {
```

Comment:
I think we have to check is operation is not ready here and reschedule task in this case